### DISCUSSION TITLE
🐛 [firebase_messaging] iOS device tokens must be no more than 400 hexadecimal characters

### DISCUSSION THREAD

**Discussion by psycura** [firebase/flutterfire#3375] [Q&A]
I'm trying to get device push token, by using method:
`_firebaseMessaging.getToken()`

As a result i getting string like:
"fUG5dIcN_pA:APA91bEciRwWuvTIezAKcJ5y1xz5z6BygE3YJkywdCGCFJD93NTfjARwPRommwgsfvVo2iH_qZWT7D2Lxnc69uanato1UUq-nLl5R1L0qF4exT7zjM9Wdy9Evs6h-EOBtIVv7Vv8bPE1"

When i try to use this token through Amazon Push service, i get error:
"iOS device tokens must be no more than 400 hexadecimal characters"

However when im use same token in the Firebase FCM console - the token work just fine.

So as i understand there some different format of token should be.

Anybody else stuck with this issue?

**Comment by Salakar** [comment]
This is a FCM token, my guess is Amazon Push service is expecting an APNs token rather than an FCM token.

**Reply by psycura** [reply]
Ok, and how can i get APN token with firebase_messaging?

**Reply by ramyareye** [reply]
https://rnfirebase.io/reference/messaging#getAPNSToken
